src/image/ExposureInfo.cc

```diff
@@ -160,8 +160,12 @@ ExposureInfo::_startWriteFits(afw::geom::Point2I const & xy0) const {
     data.metadata->combine(getMetadata());
 
     // In the future, we might not have exactly three image HDUs, but we always do right now,
-    // so 1=primary, 2=image, 3=mask, 4=variance, 5+=archive
-    data.metadata->set("AR_HDU", 5, "HDU containing the archive used to store ancillary objects");
+    // so 0=primary, 1=image, 2=mask, 3=variance, 4+=archive
+    //
+    // Historically the AR_HDU keyword was 1-indexed (see RFC-304), and to maintain file compatibility
+    // this is still the case so we're setting AR_HDU to 5 == 4 + 1
+    //
+    data.metadata->set("AR_HDU", 5, "One more than HDU containing the archive used to store ancillary objects");
```

Comment:
Thank you for the clarifying comment above, but I'm not so keen on the wording of the metadata comment. Maybe "HDU (1-indexed) containing ..."? Though I'm not sure about that either.

This also seems like it's going to bite us in the future (having a place where we 1-index). Do we need to maintain this backwards compatibility, or can we require reprocessing after this merge?